quark/subproject.py

```diff
@@ -136,6 +139,25 @@ def add_module(parent, name, uri, options, conf, **kwargs):
             if mod is newmodule:
                 mod.parents.add(parent)
                 if update:
+                    if clobber and os.path.exists(mod.directory):
+                        # if we have to clobber and a subproject directory
+                        # already exists, move it out of the way
+
+                        # ensure we have a backup root
+                        os.makedirs(clobber_backup_path, exist_ok = True)
+                        # find a name for the backup directory; we may have to
+                        # deal with collisions, as the .. trick allows for
+                        # duplicated basenames between subprojects
```

Comment:
If there is a duplicated dependency that uses the `..` "trick" and its basename also conflicts with another one I'm happy if the command fails, just to teach them a lesson, but I get your point.